internal/handlers/common/update_array_operators.go

```diff
@@ -91,6 +96,44 @@ func processPopArrayUpdateExpression(doc *types.Document, update *types.Document
 	return changed, nil
 }
 
+// checkUnsuitableValueError returns ErrUnsuitableValueType when part of
+// the path is used by non document value type.
+// If path is "v.foo" and
+// - doc is {v: 42} it returns ErrUnsuitableValueType, v is used by unsuitable value type.
+// - doc is {c: 10} it returns no error since the path does not exist.
```

Comment:
```suggestion
// checkUnsuitableValueError returns ErrUnsuitableValueType if 
// if finds an element that contains a non-document value. 
// If it fiends an element that does not exist, it returns nil.
//
// For example, if the path is "v.foo" and
// - doc is {v: 42}, it returns ErrUnsuitableValueType, v is used by unsuitable value type;
// - doc is {c: 10}, it returns no error since the path does not exist.
```

I'm not sure if my doc suggestion is better, so feel free to skip it

Reply:
I took part of it. Let me know what you think :pray: